metrics/registry.go

```diff
@@ -0,0 +1,31 @@
+package metrics
+
+type registry struct {
+	provider Provider
+}
+
+//Provider interface that implements metric registry types
+type Provider interface {
+	Init()
```

Comment:
```suggestion
	init()
```
this will always be invoked via registry handler